### PR TITLE
ci: move Archaeologist to GHA

### PR DESCRIPTION
**Pull request by jkleinsc**
#### Description of Change

- This PR in conjunction with https://github.com/electron/archaeologist/pull/44 will allow us to migrate from Archaeologist from CircleCI to GitHub Actions.  Additionally, this will allow us to support re-runs of Archaeologist by re-running the new Archaeologist Dig action which also provides the artifacts diff, eg see: https://github.com/electron/electron/actions/runs/10888035620?pr=43701

Successful archaeologist run here: https://github.com/electron/electron/pull/43701/checks?check_run_id=30211644858

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: .github/actions/generate-types/action.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    required: true
+runs:
+  using: "composite"
+  steps:
+  - name: Generating Types for SHA in ${{ inputs.sha-file }}
+    shell: bash
+    run: |
+      git checkout $(cat ${{ inputs.sha-file }})
+      rm -rf node_modules
+      yarn install --frozen-lockfile
```
Comment (MarshallOfSound) [inline .github/actions/generate-types/action.yml:18]:
```suggestion
      yarn install --frozen-lockfile --ignore-scripts
```

Might be faster

**Review by VerteDinde — APPROVED** [review]

**Review by codebytere — APPROVED** [review]

**Review by MarshallOfSound — APPROVED** [review]

**Pull request merged**